code/modules/projectiles/projectile.dm

```diff
@@ -199,11 +199,16 @@
 		L.on_hit(src)
 
 	var/reagent_note
-	if(reagents?.reagent_list)
+	if(length(reagents?.reagent_list) > 0)
 		reagent_note = " REAGENTS:"
 		for(var/datum/reagent/R in reagents.reagent_list)
 			reagent_note += "[R.name] ([num2text(R.volume)])"
-
+	if(istype(src, /obj/item/projectile/bullet/dart))
+		var/obj/item/projectile/bullet/dart/D = src
+		if(D.syringe && length(D.syringe.reagents?.reagent_list) > 0)
```

Comment:
Does this not just work as
`if(length(D.syringe?.reagents?.reagent_list) > 0)`
and if it doesn't, would it not have to be
`if(D.syringe?.reagents && length(D.syringe.reagents.reagent_list) > 0)`